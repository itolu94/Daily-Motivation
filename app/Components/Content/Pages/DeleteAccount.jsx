import React, {Component} from 'react';
import API from '../../../Utility/API.jsx';

export default class DeleteAccount extends Component {
    constructor(){
        super();
        this.state = {
            email: '',
            phoneNumber: '',
            response: ''
        };
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }
    handleSubmit(e){
        e.preventDefault();
        let {email, phoneNumber} = this.state;
        let data = {email, phoneNumber};
        API.deleteAccount(data, (response) =>{
            if(response.status === 200 || response.status === 404) this.setState({response: response.data});
            else this.setState({response: "Oops.. something went wrong.  Please try again later"});
        })
    }
    handleChange(e){
            this.setState({[e.target.name]: e.target.value});
    }
    render(){
        return(
            <div id='delete-account-div' className="container">
                <div className="row pages-div">
                    <div className="col m8 offset-m2 center-align">
                        <h3 id='delete-account-header'>To delete your account, please enter your email address and phone number</h3>
                        <p>{this.state.response}</p>
                        <form onSubmit={this.handleSubmit}>
                            <input
                                type='text'
                                name='email'
                                placeholder='Email Address'
                                value={this.state.email}
                                onChange={this.handleChange}
                                required
                            />
                            <input
                                type='text'
                                name='phoneNumber'
                                placeholder='Phone Number'
                                value={this.state.phoneNumber}
                                onChange={this.handleChange}
                                required
                            />
                            <input className='' type="submit"/>
                        </form>
                    </div>
                </div>
            </div>
        );
    }
}