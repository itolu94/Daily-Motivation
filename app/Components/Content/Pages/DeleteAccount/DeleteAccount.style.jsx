import styled from 'styled-components';

const DeleteAccountDiv = styled.div`
    background-color: powderblue;
    color: black;
    margin-top: 10%;
    padding: 10px 0;
    border-radius: 15px;
    border: 5px white solid
`;

const DeleteAccountHeader = styled.h3`
    font-size: 1.92rem;
    margin:0;
    padding:0;
    height:100%;
`;

export {DeleteAccountHeader, DeleteAccountDiv};
