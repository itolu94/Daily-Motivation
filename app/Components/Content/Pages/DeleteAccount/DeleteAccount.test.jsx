import React from 'react';
import { shallow } from 'enzyme'
import DeleteAccount from './DeleteAccount.jsx';


test("Header component", () => {
    const wrapper = shallow(<DeleteAccount/>);
    expect(wrapper).toMatchSnapshot();
});
