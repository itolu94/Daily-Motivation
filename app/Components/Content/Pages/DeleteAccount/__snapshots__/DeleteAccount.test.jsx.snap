// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `
<styled.div
  className="container"
>
  <div
    className="row pages-div"
  >
    <div
      className="col m8 offset-m2 center-align"
    >
      <styled.h3
        id="delete-account-header"
      >
        To delete your account, please enter your email address and phone number
      </styled.h3>
      <p />
      <form
        onSubmit={[Function]}
      >
        <input
          name="email"
          onChange={[Function]}
          placeholder="Email Address"
          required={true}
          type="text"
          value=""
        />
        <input
          name="phoneNumber"
          onChange={[Function]}
          placeholder="Phone Number"
          required={true}
          type="text"
          value=""
        />
        <input
          className=""
          type="submit"
        />
      </form>
    </div>
  </div>
</styled.div>
`;
